### PR TITLE
reorder HipSpecialFunctions.h inclusion so other headers can reach it…

### PR DESCRIPTION
**Pull request by sunway513**
… for HIP GPU path

### CODE REVIEW

File: Eigen/Core [hunk 1]
```diff
@@ -15,7 +15,7 @@
 #include "src/Core/util/DisableStupidWarnings.h"
 
 // Handle HIPCC/CUDA/SYCL
-#if defined(__HCC__) || defined(__NVCC__) || defined(__HIPCC__) || defined(__SYCL_DEVICE_ONLY__)
+#if defined(__HIPCC__) || defined(__SYCL_DEVICE_ONLY__)
```
Comment (whchung) [inline Eigen/Core]: @sunway513 do we want to retain `defined(__NVCC__)`?

File: unsupported/Eigen/src/SpecialFunctions/SpecialFunctionsFunctors.h [hunk 2]
```diff
@@ -219,6 +219,7 @@ template<typename Scalar> struct scalar_erfc_op {
   typedef typename packet_traits<Scalar>::type Packet;
   EIGEN_DEVICE_FUNC inline Packet packetOp(const Packet& a) const { return internal::perfc(a); }
 };
+
```
Comment (whchung) [inline unsupported/Eigen/src/SpecialFunctions/SpecialFunctionsFunctors.h]: @sunway513 pls remove this extra line.

File: unsupported/Eigen/src/SpecialFunctions/arch/HIP/HipSpecialFunctions.h [hunk 3]
```diff
@@ -18,7 +18,6 @@ namespace internal {
 // introduce conflicts between these packet_traits definitions and the ones
 // we'll use on the host side (SSE, AVX, ...)
 #if defined(__HIPCC__) && defined(EIGEN_USE_GPU)
-
```
Comment (whchung) [inline unsupported/Eigen/src/SpecialFunctions/arch/HIP/HipSpecialFunctions.h]: @sunway513 pls put this line back.

File: Eigen/Core (hunk 1 above)
Comment (sunway513) [inline Eigen/Core]: __HIPCC__ covers both __NVCC__ and __HCC__, so it's sufficient to only check against __HIPCC__. Also noted we only keep HIP backend in hipeigen as for now.

Comment (whchung) [inline Eigen/Core]: i see. then let's keep it this way.

**Pull request merged**